File: app/src/main/java/com/liyi/sample/mvvm/LoginLgc.java
package com.liyi.sample.mvvm;

import android.text.TextUtils;

public class LoginLgc {

    public void login(String phoneNo, String password, ApiListener callback) {
        if (!TextUtils.isEmpty(phoneNo) && !TextUtils.isEmpty(password)) {
            if (phoneNo.equals("18013863380") && password.equals("123456")) {
                callback.onSuccess();
            } else {
                callback.onFailure();
            }
        }
    }
}

File: module_login/src/main/java/indi/liyi/module/login/LoginLgc.kt
package indi.liyi.module.login

import android.text.TextUtils
import indi.liyi.common.utils.http.HttpCallback
import org.json.JSONObject

class LoginLgc {

    fun login(phoneNo: String, password: String, callback: HttpCallback<JSONObject>) {
        if (!TextUtils.isEmpty(phoneNo) && !TextUtils.isEmpty(password)) {
            if (phoneNo == "18013863380" && password == "123456") {
                callback.onSuccess(null, null)
            } else {
                callback.onFailure(null, null)
            }
        }
    }
}